sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -659,6 +659,12 @@ case class Aggregate(
     val nonAgg = aggregateExpressions.filter(_.find(_.isInstanceOf[AggregateExpression]).isEmpty)
     getAllValidConstraints(nonAgg)
   }
+
+  // Whether this Aggregate operator is equally the Distinct operator.
+  private[sql] def isEquallyDistinct: Boolean = {
+    groupingExpressions.size == aggregateExpressions.size &&
```

Comment:
It seems that we don't support the following case. Shall we have a test case for the following?
```scala
scala> sql("select distinct a, a from t1 left join t2 on false").explain
== Physical Plan ==
AdaptiveSparkPlan isFinalPlan=false
+- HashAggregate(keys=[a#31], functions=[])
   +- Exchange hashpartitioning(a#31, 200), ENSURE_REQUIREMENTS, [id=#194]
      +- HashAggregate(keys=[a#31], functions=[])
         +- Project [a#31, a#31]
            +- BroadcastNestedLoopJoin BuildRight, LeftOuter, false
               :- Scan hive default.t1 [a#31], HiveTableRelation [`default`.`t1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#31], Partition Cols: []]
               +- BroadcastExchange IdentityBroadcastMode, [id=#189]
                  +- Scan hive default.t2 HiveTableRelation [`default`.`t2`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [b#32], Partition Cols: []]
```

Reply:
I think we only need `aggregateExpressions` to only contains grouping columns. @wangyum can you refine it?